mathesar_ui/src/pages/schemas/schema-row/SchemaRow.svelte

```diff
@@ -35,9 +36,14 @@
       },
     });
   }
+
+  function handleSchemaSwitch() {
+    $currentSchemaId = schema.id;
+    router.goto(`/${$currentDBName}/${schema.id}`);
+  }
 </script>
 
-<div class="schema-row">
+<div class="schema-row" on:click|self={handleSchemaSwitch}>
```

Comment:
An anchor tag is better suited here instead of a div, inorder to provide better accessibility.

This will remove the need to directly access the router and setting currentSchema store. It would introduce an issue which clicking on a child might trigger the link which should be overcome by preventing default behavior when the target is not the anchor itself.

Reply:
I have made the changes you requested.